Etg.Yams/Storage/IYamsRepository.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Threading.Tasks;
+using Etg.Yams.Application;
+using Etg.Yams.Storage.Config;
+
+namespace Etg.Yams.Storage
+{
+    public interface IYamsRepository
```

Comment:
> @remusrg IApplicationRepository or IDeploymentRepository both sound good to me. Yams in front is redundant due to namespacing. I don't like the current one :)

Hahah :) I agree that Yams prefix is redundant.

> @ReubenBond Both names sound good to me. If we want a tie-breaker, I vote IDeploymentRepository

+1 for `IDeploymentRepository`. It's consistent with `DeploymentConfig`.